### PR TITLE
Chat and Sound improvements

### LINKED ISSUE #352: Disable admin chat sound
Direct messages leaving a ping-sound makes total sense.  That's because someone is specifically asking YOU and YOU alone.  But for admin chat, lots of times there's active discussions (at least within stratus event servers) and if you're playing in the match, or anything where you don't care about the ongoing discussion, it's super annoying to hear PING PING PING PING, when it doesn't relate to you, unlike messages where it always does concern you.  I know you can toggle sounds, but afaik that does both of them, and it would make sense for most people to have admin chat w/o sound and DMs with sound.

### PR DESCRIPTION
**Pull request by applenick**
# Chat and Sound improvements

This resolves #352 by allowing users to toggle their sound settings to disallow extra sound effects such as admin chat and reports. 

### All Changes Made:
- Adjusted setting values for sounds to `ALL`, `MESSAGES`, `NONE`.
  - `ALL` receive all sounds
  - `MESSAGES` only `/msg` sounds are played
  - `NONE` no sound effects play
- ~Color change for Admin chat to differentiate it from direct messages.~
- Changed DM formatting (see below)
- Lowered pitch for admin chat sound 
  -  New sound is just a tad bit different so should help tell the difference between different alerts
- Added a new `/say (team) (message)` command to allow staff to speak in team chats while observing. 

**Update: screenshot outdated, see below for new info.**

Signed-off-by: applenick <applenick@users.noreply.github.com>

### CODE REVIEW

**Comment by Xerocoles** [comment]
You should use the lighter-red instead of the dark red, cause that's the same colour as the red on most maps' xmls, so that would blend in with chat more unless you do the lighter-red (which looks totally fine anyways)

**Comment by applenick** [comment]
@Xerocoles good idea.

Updated format:
Screen Shot 2020-03-14 at 10 28 12 AM

**Comment by Brottweiler** [comment]
Not sure I want `[A]` to be red lol, I don't see how it was a problem for it for be orange like before.

**Comment by Xerocoles** [comment]
I'm realizing light red could also potentially be not ideal due to mod flairs (which is a lot of the people using /a) being the same colour. That being said, I think it still works fine, and it's defo better than dark red.

But do NOT go back to orange please, it was definitely a problem.  You couldn't instantly tell which it was -- sometimes I'd have two conversations going on, one in admin chat, one in PMs, and there were _multiple_ times where me or someone else would be responding in the wrong format (ie, another staff member pms you, but you see its a staff and you respond in /a, which includes the rest of staff who didnt see the first part of that convo.  That's super easily fixed, you just have two diff colours for A or PM.  I think light red is fine, but you could do a ton of different colours)

**Comment by applenick** [comment]
@Brottweiler I have to agree with @Xerocoles. Seeing what it looks like with a separate formatting, it’s a wonder why it wasn’t like this from the beginning. Plus, it’s only people with permission who will ever see this formatting. I suggested that too since if it were say the DM color being altered, then it would cause more confusion since people are already used to the gold for DM colors. 

If you dislike the light red color, I’d be happy to alter it to whatever color you prefer. Though personally I like the light red 🤷‍♂

**Comment by Brottweiler** [comment]
What about aqua? Or would that not stand out as much?

**Comment by applenick** [comment]
Screen Shot 2020-03-15 at 9 47 46 AM
Let me know which you like best.

**Comment by applenick** [comment]
Screen Shot 2020-03-15 at 10 43 24 AM
Updated the private message format, based off of the OCN private message format provided by @Pablete1234. And now the admin chat can stay gold! 🎉 

I think this looks good, however I wonder if the `( )` should be formatted with a dark gray?

@Brottweiler & @Xerocoles any thoughts?

**Comment by Brottweiler** [comment]
You might be able to skip the brackets entirely and also not make it italics?

**Comment by applenick** [comment]
I like the idea of no brackets, ever cleaner! I kinda like the italics but I’ll try it without and see how it looks.

**Comment by applenick** [comment]
@Brottweiler here is the suggested format.

Screen Shot 2020-03-15 at 10 52 39 AM

**Comment by Brottweiler** [comment]
That looks good. I am indifferent if you choose to use italics or not, though, but it definitely looks better without brackets.

**Comment by applenick** [comment]
Alrighty I decided to add back the italics, personally feel it looks better. 
Screen Shot 2020-03-15 at 12 45 55 PM

Let me know if there are any other changes 😄

**Comment by Pablete1234** [comment]
Italics look better

File: core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 1]
```diff
@@ -217,6 +218,12 @@ public void sendReply(Match match, Audience audience, MatchPlayer sender, @Text
     sendDirect(match, sender, receiver.getBukkit(), message);
   }
 
+  private static PersonalizedText formatPrivateMessage(String key) {
+    return new PersonalizedText(
+        new PersonalizedTranslatable(key).getPersonalizedText().color(ChatColor.GRAY).italic(true),
+        new PersonalizedText(" {0}: {1}"));
```
Comment (TheMolkaPL) [inline core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: I think this should be translatable too.

Comment (applenick) [reply core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: @TheMolkaPL In a previous PR(not sure which of mine) I’m pretty sure someone stated they would rather keep formatting “styles" outside of the translation files. I’m been trying to adhere to that. Do you think this situation would be acceptable to be translatable?

Comment (TheMolkaPL) [reply core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: In Arabic this should be `TEST :applenick3 [A]` since they read right-to-left.

Comment (Pablete1234) [reply core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: Excuse me, but i'm not sure this makes much sense. Minecraft doesn't support right-to-left to start with, and it's not like changing order of parameters in formatting is the way that you apply RTL

**Comment by applenick** [comment]
I worked on cleaning up some things in ChatDispatcher, such as adding some missing translations. 

The new command I’ve added, `/say (team) (message)` allows for staff members to send messages (while observing) to other teams. 

Screenshot of a few formatting things:
Screen Shot 2020-03-19 at 4 59 31 PM
As you can see, observing staff will be able to speak in individual team chats to notify teams of certain things. This should prove useful say if a player is spamming in `/t` and instead of a staff having to say a message in global, they can tell that player in their team chat. 
Also pictured are the final formats for [A] and /msg from previous discussion, in case anyone wanted to see.

**Comment by applenick** [comment]
@Electroid I’m moving this out of Draft as I can’t think of anything else to change with this. Let me know if there are any changes you would like and I’ll take care of them as soon as I can.

**Comment by applenick** [comment]
@Pablete1234 I agree, a new setting system would really make things easier 😆 

It’s tricky, as I’m trying to have it so users (not staff) are unable to view any setting that does not relate to them. Otherwise a simple `/toggle adminsounds on/off` would have worked very well to resolve this xD So atm, all extra sounds such as raindrops/report sounds are disabled when set to `messages`. I feel this is acceptable for the short-term, as I believe most people will either want all sounds or simply none. 

Let me know if there are any changes regarding this you feel would be wise to adjust, otherwise thanks for the feedback!

**Comment by Brottweiler** [comment]
>as I’m trying to have it so users (not staff) are unable to view any setting that does not relate to them

FWIW, when a player `/toggle chat` they see can switch to admin chat setting, which obviously doesn't do anything for them, but they still see it as an option currently.

File: core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 2]
```diff
@@ -172,54 +188,91 @@ public void sendDirect(Match match, MatchPlayer sender, Player receiver, @Text S
...
+        match,
+        player,
+        message,
+        party.getChatPrefix().toLegacyText() + PREFIX_FORMAT,
+        viewer ->
+            party.equals(viewer.getParty())
+                || (viewer.isObserving()
+                    && viewer.getBukkit().hasPermission(Permissions.ADMINCHAT)),
+        SettingValue.CHAT_TEAM);
   }
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: I think this is neat, but we don't need it. If staff want to say something, they can just use global, rarely do you need to address a single team.

**Pull request merged**